services/venice-controller/src/main/java/com/linkedin/venice/controller/kafka/consumer/AdminConsumptionTask.java

```diff
@@ -574,7 +580,54 @@ private void executeMessagesAndCollectResults() throws InterruptedException {
             int perStorePendingMessagesCount = storeQueue == null ? 0 : storeQueue.size();
             pendingAdminMessagesCount += perStorePendingMessagesCount;
             storesWithPendingAdminMessagesCount += perStorePendingMessagesCount > 0 ? 1 : 0;
-            if (e instanceof CancellationException) {
+
+            // Check if the cause is VeniceNoStoreException
+            Throwable cause = e.getCause();
```

Comment:
In the future, could we try splitting long methods into smaller ones? Adding more code to already large methods makes readability harder.